pkg/csi_driver/controller.go

```diff
@@ -446,59 +443,59 @@ func getTierFromParams(params map[string]string) string {
 	return defaultTier
 }
 
-func invalidCapacityRange(capRange *csi.CapacityRange, tier string)  error {
+// validator function to check for invalid capacity size requests
+func invalidCapacityRange(capRange *csi.CapacityRange, tier string) error {
 	validRange := provisionableCapacityForTier(tier)
 
-	minRequired := capRange.GetRequiredBytes()
-	minSet := minRequired > 0
-	maxRequested := capRange.GetLimitBytes()
-	maxSet := maxRequested > 0
-
+	requiredCap := capRange.GetRequiredBytes()
+	requireSet := requiredCap > 0
+	limitCap := capRange.GetLimitBytes()
+	limitSet := limitCap > 0
 
-
-	if maxSet && minSet && maxRequested < minRequired {
-		return fmt.Errorf("limit bytes %v is less than required bytes %v", maxRequested, minRequired)
+	if limitSet && requireSet && limitCap < requiredCap {
+		return fmt.Errorf("limit bytes %v is less than required bytes %v", limitCap, requiredCap)
 	}
 
-	if minSet{
-		if minRequired > validRange.max {
-			return fmt.Errorf("request bytes %v is more than maximum instance size bytes %v for tier %s", minRequired, validRange.max, tier)
+	if requireSet {
+		if requiredCap > validRange.max {
+			return fmt.Errorf("request bytes %vTiB is more than maximum instance size bytes %vTiB for tier %s", requiredCap/util.Tb, validRange.max/util.Tb, tier)
```

Comment:
`requiredCap/util.Tb` will round to an int IIUC, we might be returning incorrect info.